### ISSUE TITLE
Unable to submit workflow due to validation done on variables taken from inputs

### ISSUE THREAD

**Issue by vboulineau** [argoproj/argo-workflows#1984]
Checklist:

 * [x] I've included the version.
 * [x] I've included reproduction steps.
 * [x] I've included the workflow YAML.
 * [x] I've included the logs.
 
**What happened**:
Upgrading from v2.1.1 -> v2.4.3, we are not able to submit our workflow anymore, due to the added checks in the `workflow/validate/validate.go` and because we use input values as value to these fields.

Debugging, I see that in this configuration we reach the validation code with "placeholder" string as value of the `action` and `manifest` fields and, of course, it cannot pass the validate step.
See:
https://github.com/argoproj/argo/blob/v2.4.3/workflow/validate/validate.go#L454-L465

However with patched code to allow placeholders in these fields, workflow works as before (see below)

**What you expected to happen**:
Support usage of placeholders in the validate phase.

**How to reproduce it (as minimally and precisely as possible)**:

```
apiVersion: argoproj.io/v1alpha1
kind: Workflow
metadata:
  generateName: argo-datadog-agent-
spec:
  entrypoint: main
  arguments:
    parameters:
    - name: kube-state-metrics-deployment
      value: |
        apiVersion: apps/v1
        kind: Deployment
        metadata:
          name: kube-state-metrics
          namespace: kube-system
        spec:
          selector:
            matchLabels:
              k8s-app: kube-state-metrics
          replicas: 1
          template:
            metadata:
              labels:
                k8s-app: kube-state-metrics
            spec:
              serviceAccountName: kube-state-metrics
              containers:
              - name: kube-state-metrics
                image: quay.io/coreos/kube-state-metrics:v1.3.1
                ports:
                - name: http-metrics
                  containerPort: 8080
                - name: telemetry
                  containerPort: 8081
                readinessProbe:
                  httpGet:
                    path: /healthz
                    port: 8080
                  initialDelaySeconds: 5
                  timeoutSeconds: 5

    - name: kube-state-metrics-service
      value: |
        apiVersion: v1
        kind: Service
        metadata:
          name: kube-state-metrics
          namespace: kube-system
          labels:
            k8s-app: kube-state-metrics
          annotations:
            prometheus.io/scrape: 'true'
        spec:
          ports:
          - name: http-metrics
            port: 8080
            targetPort: http-metrics
            protocol: TCP
          - name: telemetry
            port: 8081
            targetPort: telemetry
            protocol: TCP
          selector:
            k8s-app: kube-state-metrics

  templates:
  - name: manifest
    inputs:
      parameters:
      - name: action
      - name: manifest
    resource:
      action: "{{inputs.parameters.action}}"
      manifest: "{{inputs.parameters.manifest}}"

  - name: main
    inputs:
      parameters:
      - name: kube-state-metrics-deployment
      - name: kube-state-metrics-service
    steps:
    - - name: kube-state-metrics-setup
        template: manifest
        arguments:
          parameters:
          - name: action
            value: "apply"
          - name: manifest
            value: "{{item}}"
        withItems:
        - "{{inputs.parameters.kube-state-metrics-deployment}}"
        - "{{inputs.parameters.kube-state-metrics-service}}"
```

**Anything else we need to know?**:

I've rebuild argo locally with this quick patch:
https://gist.github.com/vboulineau/e2b561150128913f120cead69a96fbf0

And it works properly again.

I was not able to find a workaround that would allow this syntax (or with minor modifications) to work with argo 2.4.3, but maybe I missed something.

**Environment**:
- Argo version:
```
2.4.3
```
- Kubernetes version :
```
1.16
```

**Other debugging information (if applicable)**:
```
./argo lint test-workflow.yaml
FATA[0000] test-workflow.yaml: templates.main.steps[0].kube-state-metrics-setup templates.manifest.resource.action must be one of: get, create, apply, delete, replace, patch
```

Submitting the workflow triggers the same issue of course, and after this one (if I hardcode action for instance), we have the next one triggered:
```
templates.%s.resource.manifest must be a valid yaml
```

**Comment by ptemmer** [comment]
+1

**Comment by simster7** [comment]
Will take a look at this

**Comment by simster7** [comment]
Hi @vboulineau, thanks for this excellent bug report! I saw your patch and it looks good to me. Would you like to open a PR with your changes?

**Comment by vboulineau** [comment]
Here you go: https://github.com/argoproj/argo/pull/2028

**Comment by vboulineau** [comment]
@simster7 Thank your for the merge. I don't how your releases work, can we expect to have a patch release on 2.4 with the fix, or will it be for 2.5?

**Comment by simster7** [comment]
This will be on `v2.5-rc1` which is expected to be released today
